### PR TITLE
Update Angular getting started guide

### PR DESCRIPTION
**Pull request by wlee221**
_Issue #, if available:_

_Description of changes:_ This updates getting started guide. Fixes some bugs that I've found along the way:

- Components were not styled at all, even including the Authenticator.
  - Solution: import `theme.css` from Amplify UI. This is a required step for Authenticator anyway.
- A lot of code snippet was not TypeScript strict mode compliant. Angular 12 turns on strict mode by default.
  - Solution: strictly type everything 😈 
- The sample app will **not** hide the protected component even if user hasn't authenticated. 
  - Solution: abstract API logic into its own Angular component, and put it inside the Authenticator. This is the standard Angular practice and recommended Authenticator usage.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wlee221** [comment]
Just running through the changes to see if everything works as expected. Will mark this for review once it's done.

**Review by rachelnabors — DISMISSED** [review]

File: src/fragments/start/getting-started/angular/hosting.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    // ...
+  ]
+}
+```
+</Callout>
+
+Congratulations, your app is online!
+
+![image](/images/browser-published-updated.png))
```
Comment (timngyn) [inline src/fragments/start/getting-started/angular/hosting.mdx:44]: I think there is an extra `)` on this line and so it shows up in the page as a stray character

Comment (wlee221) [reply src/fragments/start/getting-started/angular/hosting.mdx:44]: TY!
```suggestion
![image](/images/browser-published-updated.png)
```

**Comment by rachelnabors** [comment]
2020_me_thanks
Really appreciate your putting time on upleveling our Angular docs!

Closes #4616 #4551

**Review by rachelnabors — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**